content/templates.xqm

```diff
@@ -342,8 +342,12 @@ declare %private function templates:cast($values as item()*, $targetType as xs:s
             switch ($targetType)
                 case "xs:string" return
                     string($value)
-                case "xs:integer" case "xs:int" case "xs:long" return
+                case "xs:integer"
                     xs:integer($value)
+                case "xs:int"
+		    xs:int($value)
+		case "xs:long"
+		    xs:long($value)
```

Comment:
Looks like what @line-o fixed in #19 was the syntax error here - a missing `return` clause in these newly introduced `case` clauses.